File: src/Domain/Enums/BookFileType.cs
using BookManager.Domain.Exceptions;
using Tapper;

namespace BookManager.Domain.Enums;

[TranspilationSource]
public enum BookFileType
{
    Pdf,
    Epub
}

public static class DocumentFileTypeUtils
{
    public static BookFileType GetFileType(string filename)
        => Path.GetExtension(filename).ToLower() switch
        {
            ".pdf" => BookFileType.Pdf,
            ".epub" => BookFileType.Epub,
            _ => throw new UnsupportedFileTypeException()
        };
}

File: web/src/app/core/dtos/book-file-type.ts
import { BookFileType } from "./BookManager.Domain.Enums";

export function getBookFileType(fileTypeString: string): BookFileType {
  switch (fileTypeString) {
  case 'application/pdf':
  case '.pdf':
    return BookFileType.Pdf;
  case 'application/epub':
  case '.epub':
    return BookFileType.Epub;
  default:
    throw new Error(`This file type is not supported! ${fileTypeString}`);
  }
}